client/src/components/GameOverModal.js

```diff
@@ -0,0 +1,104 @@
+import React, {  useContext,  } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { Modal, Button, Typography, Box } from '@material-ui/core/';
+import { MatchContext } from '../ContextProvider/match';
+import { useHistory } from 'react-router';
+
+const useStyles = makeStyles({
+  modal: {
+    width: '20%',
+    height: '30%',
+    margin: 'auto',
+    backgroundColor: '#2196f3',
+    boxShadow:
+      '0 0 0 1px rgba(0, 0, 0, 0.1), 0 2px 4px 1px rgba(0, 0, 0, 0.18)',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    position: 'absolute',
+    borderRadius: '10px',
+  },
+  container: {
+    display: 'flex',
+    margin: 'auto',
+    justifyContent: 'center',
+    borderRadius: '10px',
+    textAlign: 'center',
+    backgroundColor: 'white',
+    width: '100%',
+    height: '100%',
+    padding: '50px',
+    flexDirection: 'column',
+  },
+  header: {
+    marginBottom: '20px',
+  },
+  input: {
+    width: '100%',
+    marginBottom: '20px',
+  },
+  button: {
+    marginLeft: '10px',
+  },
+  blue: {
+    color: '#2196f3',
+  },
+  red: {
+    color: '#ff5e62',
+  },
+});
+
+const GameOverModal = ({ open }) => {
+  const classes = useStyles();
+  const { matchState } = useContext(MatchContext);
+  const history = useHistory();
+
+  const handleClose = () => {
+    return history.push('/');
```

Comment:
Should this redirect them to the New Game page instead of the Landing Page?

Reply:
By redirecting them to the landing page, they automatically get directed to the home page if a valid httponly cookie is present. Otherwise they get directed to login. I see it can be an extra step and will make the change

I understand it should be set up such a way that whichever page user accesses, if httponly cookie is not valid they get redirected to login. This is only working for home ATM. We still need to implement it.